python/hsfs/feature_view.py

```diff
@@ -681,13 +683,53 @@ def _update_with_vector_db_result(self, vec_server, entry, passed_features):
 
     def find_neighbors(
         self,
-        embedding,
-        feature=None,
-        k=10,
-        filter=None,
-        min_score=0,
+        embedding: List[Union[int, float]],
+        feature: Optional[Feature] = None,
+        k: Optional[int] = 10,
+        filter: Optional[Union[Filter, Logic]] = None,
+        min_score: Optional[float] = 0,
         external: Optional[bool] = None,
-    ):
+    ) -> List[List[Any]]:
+        """
+        Finds the nearest neighbors for a given embedding in the vector database.
+
+        # Arguments
+            embedding: The target embedding for which neighbors are to be found.
+            feature: The feature used to compute similarity score. Required only if there
+            are multiple embeddings (optional).
+            k: The number of nearest neighbors to retrieve (default is 10).
+            filter: A filter expression to restrict the search space (optional).
+            min_score: The minimum similarity score for neighbors to be considered (default is 0).
+
+        # Returns
+            A list of feature values
+
+        !!! Example
+            ```
+            embedding_index = EmbeddingIndex()
+            embedding_index.add_embedding(name="user_vector", dimension=3)
+            fg = fs.create_feature_group(
+                        name='air_quality',
+                        embedding_index=embedding_index,
+                        version=1,
+                        primary_key=['id1'],
+                        online_enabled=True,
+                    )
+            fg.insert(data)
+            fv = fs.create_feature_view("air_quality", fg.select_all())
+            fv.find_neighbors(
+                [0.1, 0.2, 0.3],
+                k=5,
+            )
+
+            # apply filter
```

Comment:
Could you also have an example where you supply the feature name?

Reply:
I added feature as parameter and marked it as optional, otherwise having another example take too much space.